core/src/main/scala/org/apache/spark/sql/crossdata/execution/datasources/ddl.scala

```diff
@@ -273,6 +274,34 @@ private[crossdata] case class AddJar(jarPath: String)
   }
 }
 
+private[crossdata] case class AddApp(xdContext:XDContext, jarPath: String, className:String,aliasName:Option[String]=None)
+  extends LogicalPlan with RunnableCommand {
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    if (jarPath.toLowerCase.startsWith("hdfs://") || File(jarPath).exists) {
+      xdContext.addJar(jarPath)
+    } else {
+      sys.error("File doesn't exists or is not a hdfs file")
+    }
+    xdContext.addApp(path=jarPath, clss = className, alias=aliasName.getOrElse(jarPath))
```

Comment:
Using the jarPath as the default alias may be excessive. It would be enough with the name of the jar file with the extension removed